internals-js/src/federation.ts

```diff
@@ -1986,13 +1951,12 @@ export function buildSubgraph(
   };
   let subgraph: Subgraph;
   try {
-    const schema = typeof source === 'string'
-      ? buildSchema(new Source(source, name), buildOptions)
-      : buildSchemaFromAST(source, buildOptions)
+    const parsed = typeof source === 'string' ? parse(source) : source;
+    const schema =  buildSchemaFromAST(parsed, buildOptions)
     subgraph = new Subgraph(name, url, schema);
   } catch (e) {
     if (e instanceof GraphQLError && name !== FEDERATION_UNNAMED_SUBGRAPH_NAME) {
-      throw addSubgraphToError(e, name, ERRORS.INVALID_GRAPHQL);
+      throw addSubgraphToError(e, name);
```

Comment:
May be useful to know what the reasons for these changes are/if they have any user-visible behavior.